client/starwhale/api/_impl/model.py

```diff
@@ -0,0 +1,128 @@
+        desc: (str, optional) The description of the Starwhale Model Package.
+        project_uri: (str, optional) The project uri of the Starwhale Model Package. If the argument is not specified,
+            the project_uri is the config value of `swcli project select` command.
+
+    Examples:
+    ```python
+    from starwhale import model
+
+    from .user.code.evaluator import ExamplePipelineHandler
+
+    model.build(evaluator=ExamplePipelineHandler, models=["model/mnist.pth"])
+    ```
+
+    Returns:
+        None.
+    """
+    from starwhale.core.model.view import ModelTermView
+
+    if workdir is None:
+        workdir = Path.cwd()
+    else:
+        workdir = Path(workdir)
+
+    # TODO: remove configs and models when the model package supports the arbitrary including files feature.
+    configs = configs or []
+    models = models or []
+    name = name or workdir.name
+
+    model_yaml_fpath = _render_model_yaml(
+        evaluator=evaluator,
+        workdir=workdir,
+        name=name,
+        models=models,
+        configs=configs,
+        desc=desc,
+    )
+    yaml_name = str(model_yaml_fpath.relative_to(workdir))
+
+    with disable_progress_bar():
+        ModelTermView.build(
+            workdir=workdir,
+            project=project_uri,
+            yaml_name=yaml_name,
+        )
+
+
+def _ingest_obj_entrypoint_name(
+    obj: t.Any,
+    workdir: Path,
+) -> str:
+    source_path: t.Optional[_path_T] = inspect.getsourcefile(obj)
+    if source_path is None:
+        raise RuntimeError(f"failed to get source path for object: {obj}")
+    source_path = Path(source_path).resolve().absolute()
+    workdir = workdir.resolve().absolute()
+
+    relative_path = str(source_path.relative_to(workdir))
+    _parts = relative_path.split("/")
+    module_import_path = ".".join([p.rsplit(".", 1)[0] for p in _parts if p])
+    class_name = obj.__qualname__
+    entrypoint_name = f"{module_import_path}:{class_name}"
+
+    return entrypoint_name
+
+
+def _render_model_yaml(
+    evaluator: t.Any,
+    workdir: Path,
+    name: str,
+    models: t.List[_path_T],
+    configs: t.List[_path_T],
+    desc: str,
+) -> Path:
+    if not inspect.isclass(evaluator):
+        raise NoSupportError(f"evaluator only supports class type: {evaluator}")
+
+    def _relative(_path: _path_T) -> str:
+        _path = Path(_path).absolute()
+        return str(_path.relative_to(workdir))
+
+    config = {
+        "name": name,
+        "model": [_relative(p) for p in models if p],
+        "config": [_relative(p) for p in configs if p],
+        "run": {
+            "handler": _ingest_obj_entrypoint_name(evaluator, workdir),
+        },
+        "desc": desc,
+    }
+    yaml_fpath = workdir / SW_AUTO_DIRNAME / DefaultYAMLName.MODEL
+    ensure_dir(yaml_fpath.parent)
+    ensure_file(yaml_fpath, yaml.safe_dump(config, default_flow_style=False))
+    return yaml_fpath
```

Comment:
It seems that the build function generates a config file every time. Would it be better to put it in the tmp dir (not the `.starwhale/`?

Reply:
ok